### PR TITLE
improve polygon hole creation

### PR DESCRIPTION
**Pull request by mistafunk**

### CODE REVIEW

File: src/codec/encoder/HoudiniEncoder.h [hunk 1]
```diff
@@ -42,8 +42,7 @@ struct SerializedGeometry {
 	prtx::DoubleVector coords;
 	prtx::DoubleVector normals;
 	std::vector<uint32_t> counts;
-	std::vector<uint32_t> holeCounts;
-	std::vector<uint32_t> holeIndices;
+	uint32_t holeCount;
```
Comment (chr11115) [inline src/codec/encoder/HoudiniEncoder.h:45]: This holeCount could be a simple "hasHoles" bool

File: src/palladio/ModelConverter.cpp [hunk 2]
```diff
@@ -138,42 +133,39 @@ GA_Offset createPrimitives(GU_Detail* mDetail, GroupCreation gc, const wchar_t*
...
-			for (size_t hip = 0; hip < holeCounts[hi]; hip++, holeIndexPos++) {
-				primGroup->addIndex(holeIndices[holeIndexPos]);
-			}
-			mDetail->buildHoles(0.001f, 0.2f, 0, primGroup);
-			elemGroupTable.destroy(primGroup);
-		}
-	}
+	// -- handle polygon holes and primitive groups
+	const bool needsPrimGroup = createHoles || (gc == GroupCreation::PRIMCLS);
+	const bool keepPrimGroup = (gc == GroupCreation::PRIMCLS);
```
Comment (chr11115) [inline src/palladio/ModelConverter.cpp:138]: could reuse this bool for needsPrimGroup (reorder)

**Review by chr11115 — CHANGES_REQUESTED** [review]
Two very minor (not important) code nags. 
Just a question. Since we nolonger manually provide the holes, have you tried what happens if you have overlapping faces that are not supposed to create holes?

Also please fix the tests, since the build is failing again

File: src/codec/encoder/HoudiniEncoder.h (hunk 1 above)
Comment (mistafunk) [reply src/codec/encoder/HoudiniEncoder.h:45]: I leave it

**Comment by mistafunk** [comment]
Addressed review & rebased onto develop for PRT 3 support

**Comment by mistafunk** [comment]
this is superseded by #224

**Pull request closed without merging**